src/engine/math_util.c

```diff
@@ -53,14 +58,24 @@ u32 random_u16(void) {
     return gRandomSeed16;
 }
 
+// Generate a pseudorandom integer from -32768 to 32767.
+s16 random_s16(void) {
+    return (s16) random_u16();
+}
+
+// Generate a pseudorandom integer in the specified range.
+s16 random_range(s16 min, s16 max) {
```

Comment:
why is this set as s16s if it uses u16s? also potentially unsafe if min exceeds max due to how modulos work, and also potentially biased. this implementation should be altered

Reply:
This one I was on the fence with. While I agree implementation isn't safe, that seems largely like the user's fault if they provide a smaller maximum tbh. My bigger issue is passing values larger than 32767 or smaller than -32768, though this is still largely user fault if it happens.

Reply:
regardless it has a lot of bias, if min is set to 0 and max is set to 50k for example, it will be twice as likely to give values between 0 and 15k as it will be to give values between 15k and 50k.

Reply:
Oh that's valid actually, this should be using a random u32 instead.

Reply:
no thats not the issue imo, it should be using a random float. the safe implementation would be min + (random_float() * (max - min))

Reply:
This is still technically biased (and slower) but it's at least better dispersed, so fair.